libtransact/src/families/xo.rs

```diff
@@ -43,9 +43,9 @@ impl TransactionWorkload for XoTransactionWorkload {
     fn next_transaction(
         &mut self,
     ) -> Result<(TransactionPair, Option<ExpectedBatchResult>), InvalidStateError> {
-        let nonce = self
-            .rng
-            .sample_iter(&Alphanumeric)
+        let nonce = std::iter::repeat(())
+            .map(|()| self.rng.sample(Alphanumeric))
```

Comment:
Should this just continue to use `sample_iter`, as it does in `protocol/transaction.rs`

Reply:
As far as I can tell, this is the suggested way to do this; as previously written with sample_iter(), this code would no longer compile with this version of rand. The difference is that this code is using self.rng (because we seed it) and the protocol/transaction.rs code is using rand::thread_rng(), so the protocol/transaction.rs code is less sensitive to ownership issues.